src/aw-client.ts

```diff
@@ -44,130 +54,182 @@ class AWClient {
         }
     } = {};
 
-    constructor(clientname: string, testing: boolean, baseurl: string | undefined = undefined) {
+    constructor(clientname: string, options?: {testing?: boolean, baseURL?: string}) {
+        if (!options) {
+          options = {}
+        }
```

Comment:
Assign a default object to the `options` parameter instead.

Like this:

```ts
constructor(clientname: string, options: {testing?: boolean, baseURL?: string} = {})
```

Reply:
it's not exactly the same: if `options` is `null`, the default parameter will not be set. Only when it's undefined.

The default parameter is compiled down to `if (typeof options === 'undefined') { options = {} }`

Reply:
Why would options ever be `null`?

Reply:
If third party devs pass an invalid argument to a function that's their problem. The types are there, they should know better than to pass in `null` (and why would they? Makes no sense).

Reply:
95% of JS devs don't use types. Believe me, JS makes you do crazy stuff, I've been bitten by it a few time already: you wrap the API with another function that returns `null` in some cases and boom, you end up looking for 2h why it works in some cases and no others. Not every js library is being nice about `null` vs `undefined`.
It's a simple change that doesn't impact us in any way: the code is as readable either way and it can save some headaches so let's do it?

Reply:
I really think it's unnecessary. Why would you construct an options object by calling a function? Seems like overengineering around the flaws of JavaScript for an obscure case where a developer does something that would be extremely uncommon and bad practice anyway. 

Just use the default parameter syntax, it's really not worth the 3 extra lines in this case.

Reply:
> Why would you construct an options object by calling a function?

Why wouldn't you? I don't think you get to decide what someone is going to do in its code haha.

> bad practice anyway

Yes it's bad practice to mix `null` and `undefined` but in the past, a lot of libraries didn't really care about the difference and it's not unlikely that someone is going to use one.

> it's really not worth the 3 extra lines in this case

You know that it's going to get compiled to something longer in case of the default parameter, right? haha

I really don't think it's a good idea to optimise for code size over developer experience but whatever, it's a not hill I want to die on